setup.py

```diff
@@ -1,30 +1,82 @@
 from glob import glob
+import logging
 import os
 from setuptools import setup
 from setuptools.extension import Extension
 
-import numpy as np
+logging.basicConfig()
+log = logging.getLogger('probfit')
 
-try:
-    from Cython.Build import cythonize
 
-    USE_CYTHON = True
-except ImportError:
-    print('Cython is not available; using pre-generated C files')
-    USE_CYTHON = False
+class NumpyExtension(Extension, object):
+    """C Extension that implicitly uses numpy.
 
-ext = '.pyx' if USE_CYTHON else '.c'
-extensions = []
-for source_file in glob('probfit/*' + ext):
-    fname, _ = os.path.splitext(os.path.basename(source_file))
-    extensions.append(
-        Extension('probfit.{0}'.format(fname),
-                  sources=['probfit/{0}{1}'.format(fname, ext)],
-                  include_dirs=[np.get_include()])
-    )
+    This class can be useful because it defers the importing of the numpy
+    module until the include_dirs property is accessed. For setup.py commands
+    like egg_info, include_dirs is not queried, meaning numpy isn't required
+    and shouldn't be a dependency.
 
-if USE_CYTHON:
-    extensions = cythonize(extensions)
+    Taken from the "NumPy extensions and setup_requires" thread in the
+    distutils mailing list.
+    """
+    def __init__(self, *args, **kwargs):
+        self._include_dirs = []
+        super(NumpyExtension, self).__init__(*args, **kwargs)
+
+        # In Python 2, the property will be showed by the instance attribute
+        try:
+            del self.include_dirs
+        except AttributeError:
+            pass
+
+    @property
+    def include_dirs(self):
+        try:
+            from numpy import get_include
+        except:
+            log.error('Cannot import numpy; extension building will fail')
+            return self._include_dirs
+
+        return self._include_dirs + [get_include()]
+
+    @include_dirs.setter
+    def include_dirs(self, dirs):
+        self._include_dirs = dirs
+
+
+def get_extensions():
+    extensions = []
+    try:
+        from Cython.Build import cythonize
```

Comment:
Nooo! Setuptools has had native support for Cython for something like 5 years now.